webapp/src/containers/Rhs/index.tsx

```diff
@@ -1,17 +1,24 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 
 import usePluginApi from 'hooks/usePluginApi';
 
 import {getProjectDetailsState, getWebsocketEventState} from 'selectors';
 
+import pluginConstants from 'pluginConstants';
+
 import AccountNotLinked from './accountNotLinked';
 import ProjectList from './projectList';
 import ProjectDetails from './projectDetails';
 
 const Rhs = (): JSX.Element => {
-    const {state} = usePluginApi();
+    const {state, makeApiRequest} = usePluginApi();
     const {isConnected} = getWebsocketEventState(state);
 
+    // Check if user is connected on page reload
+    useEffect(() => {
+        makeApiRequest(pluginConstants.pluginApiServiceConfigs.getUserDetails.apiServiceName);
+    }, []);
```

Comment:
this is fine but we need to read the `isLoading` state for this API call so that way we can prevent that quick appearance of account not linked and show only linear-loader till then

you need to modify below
```
<div
            id='scrollableArea'
            className='overflow-auto height-rhs position-relative padding-16'
        >
            {isLoading && <LinearLoader>}
            {!isLoading && !isConnected && <AccountNotLinked/>}
            {
                !isLoading && isConnected && (
                    getProjectDetailsState(state).projectID ?
                        <ProjectDetails {...getProjectDetailsState(state)}/> :
                        <ProjectList/>)
            }
        </div>
```